staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -96,12 +96,25 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 		}
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		var transformers []rest.TransformFunc
-		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && mutatingAdmission.Handles(admission.Update) {
+		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
 			transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
-				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+				hasUID, err := hasUID(oldObj)
```

Comment:
I don't think the UID is any better than an empty resource version. I prefer the later. I also think to have seen that logic elsewhere before.

Reply:
IMHO, judging by UID or by ResourceVersion are not really different at least for now. What we are actually doing is judging is `oldObj` is created by reflect and zero-valued.

Reply:
maybe just give the variable a better name: isZeroObject

Reply:
Done.